dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskInstanceServiceImpl.java

```diff
@@ -200,4 +205,28 @@ public Result queryTaskListPaging(User loginUser,
         }
         return result;
     }
+
+    @Override
+    public Result<Object> queryRankExecuteTime(User loginUser, long projectCode, Integer top, ExecutionStatus stateType, String startTime, String endTime) {
+        Result result = new Result();
+        Project project = projectMapper.queryByCode(projectCode);
+        //check user access for project
+        Map<String, Object> checkResult = projectService.checkProjectAndAuth(loginUser, project, projectCode);
+        Status status = (Status) checkResult.get(Constants.STATUS);
+        if (status != Status.SUCCESS) {
+            putMsg(result,status);
+            return result;
+        }
+
+        String orderBy = "";
+        if (driverClassName.contains("postgresql")) {
```

Comment:
Unified sql needs to avoid using different database dateTime function.  It's seem have to add a column named 'duration'  of table 't_ds_task_instance' and maintain it when update 'end_time'. 
I'll figurate out all related interface and maintain the 'duration' column.